controllers/kuadrant_controller.go

```diff
@@ -343,46 +295,56 @@ func (r *KuadrantReconciler) registerExternalAuthorizerOSSM(ctx context.Context,
 	smcp := &maistrav2.ServiceMeshControlPlane{}
 
 	smcpKey := client.ObjectKey{Name: controlPlaneProviderName(), Namespace: controlPlaneProviderNamespace()}
-	if err := r.Client().Get(ctx, smcpKey, smcp); err != nil {
+	if err := r.GetResource(ctx, smcpKey, smcp); err != nil {
 		logger.V(1).Info("failed to get servicemeshcontrolplane object", "key", smcp, "err", err)
 		return err
 	}
+	smcpWrapper := istio.NewOSSMControlPlaneWrapper(smcp)
+	kuadrantAuthorizer := common.NewKuadrantAuthorizer(kObj.Namespace)
 
-	if smcp.Spec.TechPreview == nil {
-		smcp.Spec.TechPreview = maistrav1.NewHelmValues(nil)
-	}
-
-	var meshConfig *istiomeshv1alpha1.MeshConfig
-
-	if conf, found, err := smcp.Spec.TechPreview.GetMap("meshConfig"); err != nil {
+	hasKuadrantAuthorizer, err := common.HasKuadrantAuthorizer(smcpWrapper, *kuadrantAuthorizer)
+	if err != nil {
 		return err
-	} else if found {
-		meshConfigStruct, err := structpb.NewStruct(conf)
+	}
+	if !hasKuadrantAuthorizer {
+		err = common.RegisterKuadrantAuthorizer(smcpWrapper, kuadrantAuthorizer)
 		if err != nil {
 			return err
 		}
-		meshConfig, _ = meshConfigFromStruct(meshConfigStruct)
-	} else {
-		meshConfig = &istiomeshv1alpha1.MeshConfig{}
+		logger.Info("adding external authorizer to OSSM meshconfig")
+		if err := r.UpdateResource(ctx, smcpWrapper.GetConfigObject()); err != nil {
+			return err
+		}
 	}
-	extensionProviders := extensionProvidersFromMeshConfig(meshConfig)
 
-	if hasKuadrantAuthorizer(extensionProviders) {
-		return nil
-	}
+	return nil
+}
 
-	meshConfig.ExtensionProviders = append(meshConfig.ExtensionProviders, createKuadrantAuthorizer(kObj.Namespace))
-	meshConfigStruct, err := meshConfigToStruct(meshConfig)
-	if err != nil {
-		return err
-	}
-	smcp.Spec.TechPreview.SetField("meshConfig", meshConfigStruct.AsMap())
-	logger.Info("adding external authorizer to meshconfig")
-	if err := r.Client().Update(ctx, smcp); err != nil {
-		return err
+func (r *KuadrantReconciler) getIstioConfigObjects(ctx context.Context, logger logr.Logger) ([]common.ConfigWrapper, error) {
+	var configsToUpdate []common.ConfigWrapper
+
+	iop := &iopv1alpha1.IstioOperator{}
+	iopKey := client.ObjectKey{Name: controlPlaneProviderName(), Namespace: controlPlaneProviderNamespace()}
+	if err := r.GetResource(ctx, iopKey, iop); err != nil {
+		logger.V(1).Info("failed to get istiooperator object", "key", iopKey, "err", err)
+		if apimeta.IsNoMatchError(err) {
+			// return nil and nil if there's no istiooperator CRD, means istio is not installed
+			return nil, nil
+		} else if err.Error() != fmt.Sprintf("IstioOperator.install.istio.io \"%s\" not found", controlPlaneProviderName()) {
```

Comment:
Can we use [`apierrors.IsNotFound`](https://pkg.go.dev/k8s.io/apimachinery@v0.26.1/pkg/api/errors#IsNotFound) here?

```suggestion
		} else if !apierrors.IsNotFound(err) {
```